sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/jms/ServiceBusJmsConnectionFactoryFactory.java

```diff
@@ -40,42 +47,42 @@ <T extends ServiceBusJmsConnectionFactory> T createConnectionFactory(Class<T> fa
         try {
             T factory;
             if (properties.isPasswordlessEnabled()) {
-                String remoteUrl = String.format(AMQP_URI_FORMAT,
-                    properties.getNamespace() + "." + properties.getProfile().getEnvironment().getServiceBusDomainName(),
-                    properties.getIdleTimeout().toMillis());
-                factory = factoryClass.getConstructor(String.class).newInstance(remoteUrl);
-            } else {
-                ServiceBusConnectionString serviceBusConnectionString = new ServiceBusConnectionString(properties.getConnectionString());
-                String host = serviceBusConnectionString.getEndpointUri().getHost();
+                String hostName = properties.getNamespace() + "." + properties.getProfile().getEnvironment().getServiceBusDomainName();
+
+                TokenCredentialProvider tokenCredentialProvider = TokenCredentialProvider.createDefault(new TokenCredentialProviderOptions(properties.toPasswordlessProperties()));
+                TokenCredential tokenCredential = tokenCredentialProvider.get();
 
-                String remoteUrl = String.format(AMQP_URI_FORMAT, host, properties.getIdleTimeout().toMillis());
-                String username = serviceBusConnectionString.getSharedAccessKeyName();
-                String password = serviceBusConnectionString.getSharedAccessKey();
+                factory = factoryClass.getConstructor(TokenCredential.class, String.class, ServiceBusJmsConnectionFactorySettings.class)
+                                      .newInstance(tokenCredential, hostName, new ServiceBusJmsConnectionFactorySettings());
+
+            } else {
+                factory = factoryClass.getConstructor(String.class, ServiceBusJmsConnectionFactorySettings.class).newInstance(properties.getConnectionString(), new ServiceBusJmsConnectionFactorySettings());
+            }
+            if ("standard".equalsIgnoreCase(properties.getPricingTier())) {
+                JmsConnectionFactory jmsFactory = (JmsConnectionFactory) ReflectionUtils.getField(ServiceBusJmsConnectionFactory.class, "factory", factory);
+                jmsFactory.setExtension(JmsConnectionExtensions.AMQP_OPEN_PROPERTIES.toString(),
+                    (connection, uri) -> {
+                        Map<String, Object> properties = new HashMap<>();
+                        properties.put("com.microsoft:is-client-provider", false);
 
-                if (StringUtils.hasLength(username) && StringUtils.hasLength(password)) {
-                    factory = factoryClass.getConstructor(String.class, String.class, String.class)
-                        .newInstance(username, password, remoteUrl);
-                } else {
-                    factory = factoryClass.getConstructor(String.class).newInstance(remoteUrl);
-                }
+                        return properties;
+                    });
```

Comment:
Can we implement this using a customizer? And can we deserialize whatever proeprties set in the sdk, and remove the client-provider entry?